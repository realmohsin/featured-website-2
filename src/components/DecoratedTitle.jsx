import React from "react";
import {
  makeStyles,
  useTheme,
  Box,
  Container,
  Grid,
  Typography,
  Button,
  Paper,
} from "@material-ui/core";
import TitleDecoration from "../assets/svgs/TitleDecoration";

import clsx from "clsx";

const useStyles = makeStyles((theme) => ({
  sectionTitle: {
    margin: "0 0 12rem",
    position: "relative",
    [theme.breakpoints.down("xs")]: {
      margin: "0 0 10rem",
    },
  },
  title: {
    textShadow: "1px 1px 5px rgba(0,0,0,0.2)",
    fontSize: "4.8rem",
    fontWeight: "bold",
    color: theme.palette.primary.main,
    [theme.breakpoints.down("md")]: {
      fontSize: "4.6rem",
    },
    [theme.breakpoints.down("xs")]: {
      fontSize: "3.8rem",
    },
  },
  decoration: {
    position: "absolute",
    // top: '150%',
    // left: '73px'
    top: "150%",
    left: "50%",
    transform: "translateX(-50%)",
  },
  closer: {
    top: "130%",
  },
  smallerFont: {
    fontSize: "4.2rem",
  },
}));

const SectionTitle = ({ title, closer, classStyles, smallerFont }) => {
  const theme = useTheme();
  const classes = useStyles();
  return (
    <Box className={clsx(classes.sectionTitle, classStyles)}>
      <Typography
        variant="h2"
        className={clsx(classes.title, smallerFont && classes.smallerFont)}
        align="center"
      >
        {title}
      </Typography>
      <TitleDecoration
        color={theme.palette.secondary.main}
        className={clsx(classes.decoration, closer && classes.closer)}
      />
    </Box>
  );
};

export default SectionTitle;
