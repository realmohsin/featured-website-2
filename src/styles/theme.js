import { createMuiTheme } from '@material-ui/core/styles'

const primaryColor = '#0E5C60'
const primaryDarkColor = '#002B2E'
const primaryVariant1 = '#00464A'
const secondaryColor = '#CCAB63'
const goldVariant1 = '#E1C079'
const facebookColor = '#3C5B9B'
const googleStarColor = '#E7711B'
const fadedTextColor = '#94B7B8'

const theme = createMuiTheme({
  palette: {
    primary: {
      main: primaryColor,
      dark: primaryDarkColor
    },
    secondary: {
      main: secondaryColor
    },
    common: {
      primaryVariant1,
      goldVariant1,
      facebookColor,
      googleStarColor,
      fadedTextColor
    }
  },
  typography: {
    htmlFontSize: 10,
    fontFamily: ['Roboto', 'sans-serif'].join(', '),
    body1: {
      fontSize: '1.6rem',
      '@media (max-width: 600px)': {
        fontSize: '1.4rem'
      }
    },
    body2: {
      fontSize: '1.4rem',
      lineHeight: '1.5'
    }
  },
  overrides: {
    MuiCssBaseline: {
      '@global': {
        '*, *::before, *::after': {
          margin: 0,
          padding: 0
        },
        html: {
          fontSize: '62.5%'
        },
        a: {
          display: 'block',
          textDecoration: 'none',
          color: 'inherit'
        },
        img: {
          display: 'block'
        },
        'input, button, textarea, select': {
          display: 'block',
          font: 'inherit'
        },
        ul: {
          listStyle: 'none'
        },
        '::selection': {
          backgroundColor: primaryColor,
          color: 'white'
        },
        'a, button': {
          cursor: 'pointer'
        }
      }
    },
    MuiTypography: {
      gutterBottom: {
        marginBottom: '1.2em'
      }
    }
  },
  custom: {
    title1: {
      fontFamily: ['Piazzolla', 'serif'].join(', '),
      fontSize: '5.1rem',
      color: primaryColor,
      '@media (max-width: 600px)': {
        fontSize: '2rem'
      }
    },
    title2: {
      fontFamily: ['PT Serif', 'serif'].join(', '),
      fontSize: '4.6rem',
      color: primaryColor,
      fontWeight: 'normal',
      marginBottom: '8rem',
      textAlign: 'center'
    },
    title3: {
      fontFamily: ['PT Serif', 'serif'].join(', '),
      fontSize: '3.4rem',
      color: secondaryColor,
      fontWeight: 'normal'
    },
    title4: {
      fontSize: '2.2rem',
      color: primaryColor,
      textTransform: 'uppercase',
      wordSpacing: '0.5px'
    },
    title5: {
      fontSize: '1.5rem',
      color: secondaryColor,
      textTransform: 'uppercase',
      fontWeight: 'normal'
    },
    outlineTitle: {
      textShadow: `-1px 0 ${primaryColor}, 0 1px ${primaryColor}, 1px 0 ${primaryColor}, 0 -1px ${primaryColor}`
    },
    inSideDrawer: {
      transform: 'translateY(0)',
      opacity: 1,
      pointerEvents: 'auto',
      width: 'auto',
      height: 'auto',
      border: 'none',
      backgroundColor: 'white',
      boxShadow: 'inset 0 0 5px 2px rgba(0,0,0,.2)',
      '& a': {
        color: primaryColor,
        fontSize: '1.1rem',
        display: 'flex',
        height: '4rem',
        borderBottom: `1px solid #eee`,
        alignItems: 'center',
        justifyContent: 'space-between',
        paddingRight: '1rem',
        fontWeight: 'normal',
        paddingLeft: '3.6rem'
      }
    }
  }
})

export default theme
