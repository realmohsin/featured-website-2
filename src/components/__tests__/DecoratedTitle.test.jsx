import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "@material-ui/core/styles";
import { createMuiTheme } from "@material-ui/core/styles";
import DecoratedTitle from "../DecoratedTitle";

// Mock the TitleDecoration SVG component
jest.mock("../../assets/svgs/TitleDecoration", () => {
  return function TitleDecoration({ color, className }) {
    return <div data-testid="title-decoration" className={className} style={{ color }} />;
  };
});

const theme = createMuiTheme({
  palette: {
    primary: { main: "#0E5C60", light: "#4A8B8F" },
    secondary: { main: "#CCAB63" },
  },
  breakpoints: {
    down: () => "@media (max-width: 600px)",
  },
});

const renderWithTheme = (component) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
};

describe("DecoratedTitle Component", () => {
  it("renders with basic title", () => {
    const { container } = renderWithTheme(<DecoratedTitle title="Test Title" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with closer prop", () => {
    const { container } = renderWithTheme(
      <DecoratedTitle title="Closer Title" closer />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with smallerFont prop", () => {
    const { container } = renderWithTheme(
      <DecoratedTitle title="Smaller Font Title" smallerFont />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with custom classStyles", () => {
    const { container } = renderWithTheme(
      <DecoratedTitle title="Custom Styled Title" classStyles="custom-class" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with all props combined", () => {
    const { container } = renderWithTheme(
      <DecoratedTitle
        title="All Props Title"
        closer
        smallerFont
        classStyles="combined-class"
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
