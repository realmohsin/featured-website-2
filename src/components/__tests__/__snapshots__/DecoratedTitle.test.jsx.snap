// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DecoratedTitle Component renders with all props combined 1`] = `
<div
  class="MuiBox-root MuiBox-root-30 makeStyles-sectionTitle-25 combined-class"
>
  <h2
    class="MuiTypography-root makeStyles-title-26 makeStyles-smallerFont-29 MuiTypography-h2 MuiTypography-alignCenter"
  >
    All Props Title
  </h2>
  <div
    class="makeStyles-decoration-27 makeStyles-closer-28"
    data-testid="title-decoration"
    style="color: rgb(204, 171, 99);"
  />
</div>
`;

exports[`DecoratedTitle Component renders with basic title 1`] = `
<div
  class="MuiBox-root MuiBox-root-6 makeStyles-sectionTitle-1"
>
  <h2
    class="MuiTypography-root makeStyles-title-2 MuiTypography-h2 MuiTypography-alignCenter"
  >
    Test Title
  </h2>
  <div
    class="makeStyles-decoration-3"
    data-testid="title-decoration"
    style="color: rgb(204, 171, 99);"
  />
</div>
`;

exports[`DecoratedTitle Component renders with closer prop 1`] = `
<div
  class="MuiBox-root MuiBox-root-12 makeStyles-sectionTitle-7"
>
  <h2
    class="MuiTypography-root makeStyles-title-8 MuiTypography-h2 MuiTypography-alignCenter"
  >
    Closer Title
  </h2>
  <div
    class="makeStyles-decoration-9 makeStyles-closer-10"
    data-testid="title-decoration"
    style="color: rgb(204, 171, 99);"
  />
</div>
`;

exports[`DecoratedTitle Component renders with custom classStyles 1`] = `
<div
  class="MuiBox-root MuiBox-root-24 makeStyles-sectionTitle-19 custom-class"
>
  <h2
    class="MuiTypography-root makeStyles-title-20 MuiTypography-h2 MuiTypography-alignCenter"
  >
    Custom Styled Title
  </h2>
  <div
    class="makeStyles-decoration-21"
    data-testid="title-decoration"
    style="color: rgb(204, 171, 99);"
  />
</div>
`;

exports[`DecoratedTitle Component renders with smallerFont prop 1`] = `
<div
  class="MuiBox-root MuiBox-root-18 makeStyles-sectionTitle-13"
>
  <h2
    class="MuiTypography-root makeStyles-title-14 makeStyles-smallerFont-17 MuiTypography-h2 MuiTypography-alignCenter"
  >
    Smaller Font Title
  </h2>
  <div
    class="makeStyles-decoration-15"
    data-testid="title-decoration"
    style="color: rgb(204, 171, 99);"
  />
</div>
`;
